catalog/cataloger_diff.go

```diff
@@ -59,56 +78,79 @@ func (c *cataloger) doDiffByRelation(tx db.Tx, relation RelationType, leftID, ri
 			"left_id":       leftID,
 			"right_id":      rightID,
 		}).Debug("Diff by relation - unsupported type")
-		return nil, ErrFeatureNotSupported
+		return ErrFeatureNotSupported
 	}
 }
 
-func (c *cataloger) diffFromParent(tx db.Tx, parentID, childID int64) (Differences, error) {
+func (c *cataloger) getDiffSummary(tx db.Tx) (map[DifferenceType]int, error) {
+	var results []DiffTypeCount
+	err := tx.Select(&results, "SELECT diff_type, count(diff_type) as count FROM "+diffResultsTableName+" GROUP BY diff_type")
+	if err != nil {
+		return nil, fmt.Errorf("count diff resutls by type: %w", err)
+	}
+	m := make(map[DifferenceType]int, len(results))
+	for _, res := range results {
+		m[DifferenceType(res.DiffType)] = res.Count
+	}
+	return m, nil
+}
+
+func (c *cataloger) diffFromParent(tx db.Tx, parentID, childID int64) error {
 	// get the last child commit number of the last parent merge
 	// if there is none - then it is  the first merge
 	var maxChildMerge CommitID
 	childLineage, err := getLineage(tx, childID, UncommittedID)
 	if err != nil {
-		return nil, fmt.Errorf("child lineage failed: %w", err)
+		return fmt.Errorf("child lineage failed: %w", err)
 	}
 	parentLineage, err := getLineage(tx, parentID, CommittedID)
 	if err != nil {
-		return nil, fmt.Errorf("parent lineage failed: %w", err)
+		return fmt.Errorf("parent lineage failed: %w", err)
 	}
 	maxChildQuery, args, err := sq.Select("MAX(commit_id) as max_child_commit").
 		From("catalog_commits").
 		Where("branch_id = ? AND merge_type = 'from_parent'", childID).
 		PlaceholderFormat(sq.Dollar).
 		ToSql()
 	if err != nil {
-		return nil, fmt.Errorf("get child last commit sql: %w", err)
+		return fmt.Errorf("get child last commit sql: %w", err)
 	}
 	err = tx.Get(&maxChildMerge, maxChildQuery, args...)
 	if err != nil {
-		return nil, fmt.Errorf("get child last commit failed: %w", err)
+		return fmt.Errorf("get child last commit failed: %w", err)
 	}
 	diffFromParentSQL, args, err := sqDiffFromParentV(parentID, childID, maxChildMerge, parentLineage, childLineage).
 		Prefix(`CREATE TEMP TABLE ` + diffResultsTableName + " ON COMMIT DROP AS ").
 		PlaceholderFormat(sq.Dollar).
 		ToSql()
 	if err != nil {
-		return nil, fmt.Errorf("diff from parent sql: %w", err)
+		return fmt.Errorf("diff from parent sql: %w", err)
 	}
 	if _, err := tx.Exec(diffFromParentSQL, args...); err != nil {
-		return nil, fmt.Errorf("select diff from parent: %w", err)
+		return fmt.Errorf("select diff from parent: %w", err)
 	}
-	return diffReadDifferences(tx)
+	return nil
 }
 
-func diffReadDifferences(tx db.Tx) (Differences, error) {
+func getDiffDifferences(tx db.Tx, limit int, after string) (Differences, error) {
 	var result Differences
-	if err := tx.Select(&result, "SELECT diff_type, path FROM "+diffResultsTableName); err != nil {
+	query, args, err := psql.Select("diff_type", "path").
```

Comment:
HOPE TO HAVE THIS WORKING SOON, SO i A NOT SURE IT IS NEEDED

Reply:
I take Thursday off, So I guess it will be ready by mid next week

Reply:
Sure. Also there is DiffUncommitted - but I think the current solution should (unless you found an issue there)